_includes/archive-single.html

```diff
@@ -24,11 +24,7 @@ <h2 class="archive__item-title" itemprop="headline">
         <a href="{{ post.url | relative_url }}" rel="permalink">{{ title }}</a>
       {% endif %}
     </h2>
-    {% if include.type == "grid" %}
-      {% include post__meta.html p="post" type="grid" %}
-    {% else %}
-      {% include post__meta.html p="post" %}
-    {% endif %}
+    {% include post__meta.html type = include.type %}
```

Comment:
I haven't tested, but I'm afraid the spaces around the equal sign may lead to unexpected results. Also for the sake of consistency (nowhere else are there spaces like this in an `{% include %}` directive).

```diff
-    {% include post__meta.html type = include.type %}
+    {% include post__meta.html type=include.type %}
```